lib/ruby-mpd.rb

```diff
@@ -140,6 +140,8 @@ def disconnect
       @socket.close
     rescue Errno::EPIPE
       # socket was forcefully closed
+    ensure
+      @socket = nil
```

Comment:
I think you should just move the `reset_vars` call to here. It will clear out the variables for you.